### PR TITLE
Move to rebar3

### PR DESCRIPTION
**Pull request by marco-m**
I suggest looking at each commit on this branch, with its own comment, o understand what is going on.

This is fully tested with rebar3 eunit.

On the other hand, this PR removes rebar2 compatibility. This also explains all the changes to the Makefile: many commands do not have sense/equivalent with rebar3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,22 +5,31 @@ Erlang [Slack](http://slack.com) REST API wrapper.
 
 To use slacker, you need a team token that you can grab [here](https://api.slack.com/#auth).
 
+Then, create file `priv/token.txt` containing your token. Use `priv/token.txt.example` as template.
```
Comment (julienXX) [inline README.md]: We may want to specify that this token is only for tests.

File: test/slacker_tests.erl [hunk 2]
```diff
@@ -24,7 +24,10 @@ test_slack_is_up_and_running() ->
     ?assertEqual(true, get_val(<<"ok">>, Body)).
 
 test_unauth_get() ->
-    Token = "bad_token",
+    % The token in file token.txt.example is invalid, as we want for this test.
+    % We use read_token/1 instead than passing directly an invalid string
+    % to be sure that we can read the directory containing the token files.
+    {ok, Token} = read_token("token.txt.example"),
```
Comment (julienXX) [inline test/slacker_tests.erl]: Way better 👍

**Comment by marco-m** [comment]
Ah ok, github is confused by history rewriting, so I think we lost the comment about the README and the token. I updated file README acconrdingly.

**Comment by julienXX** [comment]
Awesome, thanks!

**Pull request merged**